conformance/lang/expressions/type_cast_expression.balt

```diff
@@ -0,0 +1,2012 @@
+             result of the type cast expression is v using singleton type.
+Labels: any, module-type-defn, singleton-type, string, type-cast-expr, union-type
+
+type MySingleton1 true;
+
+type MySingleton2 32;
+
+type MySingletonUnion MySingleton1|MySingleton2;
+
+function init() {
+    int a = 4;
+    4 a1 = <4>a;
+    io:println(a1); // @output 4
+
+    "hello"|"ballerina" b = "ballerina";
+    "ballerina" b1 = <"ballerina">b;
+    io:println(b1); // @output ballerina
+
+    MySingletonUnion c = true;
+    MySingleton1 c1 = <MySingleton1>c;
+    io:println(c1); // @output true
+
+    int d = 32;
+    MySingleton2 d1 = <32>d;
+    io:println(d1); // @output 32
+}
+
+Test-Case: output
+Description: Test if the value v of the expression belongs to the type described by the type cast expression then the 
+             result of the type cast expression is v using optional type.
+Labels: any, module-type-defn, nil-type, optional-type, singleton-type, string, type-cast-expr, union-type, 
+        value:toBalString
+
+type MyType1 true?;
+
+type MyType2 "ballerina"?;
+
+type MYTypeUnion MyType1|MyType2;
+
+function init() {
+    any a = 5;
+    int? a1 = <int?>a;
+    io:println(a1); // @output 5
+
+    string|int|() b0 = 55;
+    int? b1 = <int?> b0;
+    io:println(b1); // @output 55
+
+    MYTypeUnion c0 = true;
+    MyType1 c1 = <MyType1> c0;
+    io:println(c1); // @output true
+
+    MYTypeUnion d0 = ();
+    MyType1 d1 = <MyType1> d0;
+    io:println(d1.toBalString()); // @output ()
+
+    string? e0 = ();
+    MyType1 e1 = <MyType1>e0;
+    io:println(e1.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test if the value v of the expression belongs to the type described by the type cast expression then the 
+             result of the type cast expression is v using anydata, byte, xml and json type.
+Labels: any, array-type, anydata, byte, json, list-constructor-expr, map-type, record-type, string, type-cast-expr, 
+        union-type, xml
+
+function init() {
+    any a = 5;
+    anydata a1 = <anydata>a;
+    io:println(a1); // @output 5
+
+    anydata|json b0 = 55;
+    anydata b1 = <anydata>b0;
+    io:println(b1); // @output 55
+
+    byte|json c = 4;
+    byte c1 = <byte>c;
+    io:println(c1); // @output 4
+
+    xml|json d = 4;
+    json d2 = <(json)>d;
+    io:println(d2); // @output 4
+
+    int[3] e3 = [1, 2, 3];
+    json[3] e4 = <json[3]>e3;
+    io:println(e4); // @output [1,2,3]
+
+    record {| int field1; int[2] field2;|} f = {field1: 3, field2: [1, 2]};
+    map<json> f2 = <map<json>>f;
+    io:println(f2); // @output {"field1":3,"field2":[1,2]}
+}
+
+Test-Case: output
+Description: Test if the value v of the expression belongs to the type described by the type cast expression then the 
+             result of the type cast expression is v using by casting int variable length array to json fixed 
+             length array.
+Fail-Issue: ballerina-platform/ballerina-lang#36433
```

Comment:
This is not a bug. See https://github.com/ballerina-platform/ballerina-lang/issues/36433#issuecomment-1278824339.

Reply:
See https://github.com/ballerina-platform/ballerina-lang/issues/36433#issuecomment-1280515702. This should compile, but panic at runtime. This test needs to be a panic test.